File: app/src/main/java/com/example/rekraprima/PengajuanPerjalananDinasActivity.java
package com.example.rekraprima;

import android.app.DatePickerDialog;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputEditText;
import java.util.Calendar;

public class PengajuanPerjalananDinasActivity extends AppCompatActivity {

    private TextInputEditText txtTujuan, txtKeberangkatan, txtKembali, txtAgenda;
    private MaterialButton btnKirim, btnKembali;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.karyawan_pengajuan_perjalanan_dinas);

        // Sambungkan view — sesuaikan ID dengan yang ada di XML kamu
        txtTujuan        = findViewById(R.id.txtJudul);
        txtKeberangkatan = findViewById(R.id.txtTanggal);
        txtKembali       = findViewById(R.id.txtJenis);
        txtAgenda        = findViewById(R.id.txtPerihal);
        btnKirim   = findViewById(R.id.btnDaftar);
        btnKembali = findViewById(R.id.btnBatal);
        // SESUDAH
        boolean readOnly    = getIntent().getBooleanExtra("MODE_READ_ONLY", false);
        boolean modeManager = getIntent().getBooleanExtra("MODE_MANAGER", false);

        if (modeManager) {
            Pengajuan p = (Pengajuan) getIntent().getSerializableExtra("DATA_PENGAJUAN");
            tampilkanModeReadOnly(p);
            String id = getIntent().getStringExtra("PENGAJUAN_ID");
            ManagerModeHelper.setup(this, btnKirim, id, p.status, null);
        } else if (readOnly) {
            tampilkanModeReadOnly(null); // null = ambil dari intent di dalam method
        } else {
            setupModeEdit();
        }

        btnKembali.setOnClickListener(v -> finish());
    }

    // ─── Mode edit (pengajuan baru) ───────────────────────────────────────────

    private void setupModeEdit() {
        // Date picker untuk tanggal keberangkatan
        txtKeberangkatan.setFocusable(false);
        txtKeberangkatan.setOnClickListener(v -> showDatePicker(txtKeberangkatan));

        // Date picker untuk tanggal kembali
        txtKembali.setFocusable(false);
        txtKembali.setOnClickListener(v -> showDatePicker(txtKembali));

        btnKirim.setText("Kirim");
        btnKirim.setOnClickListener(v -> kirim());
    }

    private void kirim() {
        String tujuan    = getString(txtTujuan);
        String berangkat = getString(txtKeberangkatan);
        String kembali   = getString(txtKembali);
        String agenda    = getString(txtAgenda);

        // Validasi field tidak boleh kosong
        if (tujuan.isEmpty() || berangkat.isEmpty() || kembali.isEmpty() || agenda.isEmpty()) {
            Toast.makeText(this, "Semua field harus diisi!", Toast.LENGTH_SHORT).show();
            return;
        }

        // Simpan ke DataStore
        DataStore.getInstance().tambahPengajuan(
                "Perjalanan Dinas", tujuan, berangkat, kembali, agenda);

        // Tampilkan pesan berhasil
        Toast.makeText(this, "Pengajuan berhasil!", Toast.LENGTH_SHORT).show();

        // Kembali ke menu (activity ini selesai)
        finish();
    }

    // ─── Mode baca (dari daftar pengajuan) ───────────────────────────────────

    // SESUDAH
    private void tampilkanModeReadOnly(Pengajuan p) {
        if (p == null) p = (Pengajuan) getIntent().getSerializableExtra("DATA_PENGAJUAN");

        // Isi teks
        txtTujuan.setText(p.tujuan);
        txtKeberangkatan.setText(p.tanggalBerangkat);
        txtKembali.setText(p.tanggalKembali);
        txtAgenda.setText(p.agenda);

        // Nonaktifkan semua field agar tidak bisa diedit
        txtTujuan.setEnabled(false);
        txtKeberangkatan.setEnabled(false);
        txtKembali.setEnabled(false);
        txtAgenda.setEnabled(false);

        // Ubah tombol Kirim menjadi Terkirim (abu-abu, tidak bisa diklik)
        btnKirim.setText("Terkirim");
        btnKirim.setEnabled(false);
        btnKirim.setAlpha(0.5f);
    }

    // ─── Helper ──────────────────────────────────────────────────────────────

    private String getString(TextInputEditText field) {
        return field.getText() != null ? field.getText().toString().trim() : "";
    }

    private void showDatePicker(TextInputEditText target) {
        Calendar cal = Calendar.getInstance();
        new DatePickerDialog(this, (view, year, month, day) -> {
            // Format: DD/MM/YYYY
            target.setText(String.format("%02d/%02d/%04d", day, month + 1, year));
        }, cal.get(Calendar.YEAR), cal.get(Calendar.MONTH), cal.get(Calendar.DAY_OF_MONTH)).show();
    }
}

File: app/app/src/main/java/com/example/rekraprima/PengajuanPerjalananDinasActivity.kt
package com.example.rekraprima

import android.app.DatePickerDialog
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton
import com.google.android.material.textfield.TextInputEditText
import java.util.Calendar

class PengajuanPerjalananDinasActivity : AppCompatActivity() {

    private lateinit var dbHelper: DatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.karyawan_pengajuan_perjalanan_dinas)

        dbHelper = DatabaseHelper(this)

        val txtTujuan = findViewById<TextInputEditText>(R.id.txtJudul)
        val txtTglBerangkat = findViewById<TextInputEditText>(R.id.txtTanggal)
        val txtTglKembali = findViewById<TextInputEditText>(R.id.txtJenis)
        val txtAgenda = findViewById<TextInputEditText>(R.id.txtPerihal)

        val btnKirim = findViewById<MaterialButton>(R.id.btnDaftar)
        val btnKembali = findViewById<MaterialButton>(R.id.btnBatal)

        txtTglBerangkat.isFocusable = false
        txtTglBerangkat.isClickable = true

        txtTglKembali.isFocusable = false
        txtTglKembali.isClickable = true

        txtTglBerangkat.setOnClickListener {
            showDatePickerDialog(txtTglBerangkat)
        }

        txtTglKembali.setOnClickListener {
            showDatePickerDialog(txtTglKembali)
        }

        btnKirim.setOnClickListener {
            val tujuan = txtTujuan.text.toString().trim()
            val tglBerangkat = txtTglBerangkat.text.toString().trim()
            val tglKembali = txtTglKembali.text.toString().trim()
            val agenda = txtAgenda.text.toString().trim()

            if (tujuan.isEmpty() || tglBerangkat.isEmpty() || tglKembali.isEmpty() || agenda.isEmpty()) {
                Toast.makeText(this, "Semua data wajib diisi!", Toast.LENGTH_SHORT).show()
            } else {
                val hasil = dbHelper.insertPengajuan(tujuan, tglBerangkat, tglKembali, agenda)

                if (hasil > -1) {
                    Toast.makeText(this, "Pengajuan dinas berhasil dikirim!", Toast.LENGTH_SHORT).show()
                    finish()
                } else {
                    Toast.makeText(this, "Gagal menyimpan data ke database", Toast.LENGTH_SHORT).show()
                }
            }
        }

        btnKembali.setOnClickListener {
            finish()
        }
    }

    private fun showDatePickerDialog(editText: TextInputEditText) {
        val calendar = Calendar.getInstance()
        val year = calendar.get(Calendar.YEAR)
        val month = calendar.get(Calendar.MONTH)
        val day = calendar.get(Calendar.DAY_OF_MONTH)

        val datePickerDialog = DatePickerDialog(
            this,
            { _, selectedYear, selectedMonth, selectedDay ->
                // Format hasil tanggal menjadi DD/MM/YYYY (Bulan ditambah 1 karena indeks mulai dari 0)
                val formattedDate = String.format("%02d/%02d/%d", selectedDay, selectedMonth + 1, selectedYear)
                editText.setText(formattedDate)
            },
            year,
            month,
            day
        )
        datePickerDialog.show()
    }
}